### PR TITLE
ARM: add skeleton project with ARMv7/ARMv8 devices for add-on building

### PR DESCRIPTION
**Pull request by chewitt**
This adds an ARM project with two devices (ARMv7/ARMv8) and a minimal set of distro options with the goal of building only two sets of binary add-ons that are binary-compatible with all our ARMv7/v8 SoC devices respectively. Current use of webserver aliases and per-project or per-device overrides (some ARMv8 devices are using Allwinner/H3 add-ons for a while now) is a little haphazard so this should permit a standard approach. If it works it should also save build-time and CPU resources on Jenkins.

### CODE REVIEW

File: projects/ARM/devices/ARMv7/options [hunk 1]
```diff
@@ -0,0 +1,21 @@
+################################################################################
+# setup system defaults
+################################################################################
+
+  # The TARGET_CPU variable controls which processor should be targeted for
+  # generated code.
+    TARGET_CPU="cortex-a8"
```
Comment (lrusak) [inline projects/ARM/devices/ARMv7/options:7]: I don't think we should be targeting a cpu

Comment (HiassofT) [reply projects/ARM/devices/ARMv7/options:7]: I fully agree.

It's a bit tricky though ATM as config/arch.arm is built around the CPU (and defaults to cortex-a53 if not set) but at the end it only affects `-march` and ends up in `-mtune`.

I suggested on Slack to tackle that in a second step after we decided what we want - eg no `-mtune` at all, some typical, specific CPU for the arch (as it's now) or maybe `-mtune=generic-ARCH`

**Comment by CvH** [comment]
If nobody has concrete complaints or improvements I merge this tomorrow.

**Comment by CvH** [comment]
first tests looking great so merge it and lets see what happens :)

**Pull request merged**